doc/protocols.rst

```diff
@@ -146,3 +155,731 @@ If during the timeout another command except the status command is received then
 A list of all configurable parameters and their maximum and minimum values can be found here: `Lower Level Configuration Parameters`_.
 
 .. _Lower Level Configuration Parameters: ./configuration_parameters.html
+
+JSON Schemas
+------------
+
+In order to validate the JSON messages, eight JSON schemas have been constructed.
+When the schemas are used for validation, the keys in the JSON data can be used to select which schema to use.
+The selection can be done for instance as follows:
+
+* If the "command" key is present, use the command schema.
+* If the "timeout" key is present, use the response schema.
+* In all other cases, test for the presence of the key containing the sub-system name ("APCS", "ApSCS", etc) and use the corresponding status schema.
+
+This is the command schema.
+It looks rather complex since it covers all available commands (including the "config" command), however otherwise 24 separate schemas would need to be created.
+For a full list of the commands and their parameters, see `Lower Level Commands`_.
+
+.. _Lower Level Commands: ./commands.html
+
+.. code-block:: none
+
+    {
```

Comment:
See comment on validation_tools. When you keep schemas in files, you can put links there to schema files instead of copy&paste.

Reply:
As discussed on Slack yesterday, I will modify this as well.